.github/workflows/ci.yml

```diff
@@ -0,0 +1,72 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  verify:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 10.x
+
+      - name: Get Cypress Version
+        id: cypress-version
+        run: echo "::set-output name=version::$(yarn info cypress version --silent)"
+
+      - name: Cache Cypress
+        id: cypress-cache
+        uses: actions/cache/@v1
+        with:
+          path: .cache/cypress
+          key: ${{ runner.os }}-cypress-cache-version-${{ steps.cypress-version.outputs.version }}
+
+      - name: Cache node modules
+        id: yarn-cache
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-hash-${{ hashFiles('yarn.lock') }}
+
+      - name: Install Packages
+        if:
+          steps.yarn-cache.outputs.cache-hit != 'true' || steps.cypress-cache.outputs.cache-hit !=
+          'true'
+        run: yarn install --production=false
+        env:
+          CYPRESS_CACHE_FOLDER: .cache/cypress
+
+        # Keep steps separate for Github Actions annotation matching: https://github.com/actions/setup-node/blob/83c9f7a7df54d6b57455f7c57ac414f2ae5fb8de/src/setup-node.ts#L26-L33
+      - name: Lint
+        run: yarn lint
+
+      - name: Dependency Check
+        run: yarn depcheck
+
+      - name: Unit tests
+        run: yarn test
+
+      - name: Build Storybook
+        run: yarn build-storybook
+
+      - name: Start Server
+        run: yarn http-server docs -p 9001 & npx wait-on http://localhost:9001
+
+      - uses: chromaui/action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          appCode: dlpro96xybh
+          storybookBuildDir: docs
+          exitOnceUploaded: true
+          exitZeroOnChanges: true
```

Comment:
The chromatic CLI command doesn't detect Github Actions as a CI provider. The Github Action here was built for this purpose.

Reply:
We used to have a [separate action for chromatic](https://github.com/Workday/canvas-kit/actions?query=workflow%3AChromaticQA). I don't exactly remember why though

Reply:
I think we had separate workflows and jobs so that everything would run in parallel to be faster.

I removed the integration job and merged with the `unit_test` job since unit tests don't actually take that long. We also now do `exitOnceUploaded`, so we don't wait for Chromatic anymore. We instead wait on Chromatic to set a status on the commit - it is possible for this CI job to finish before Chromatic does, but that's okay. The PR will be blocked until Chromatic sets a passing status.

The slowest part of all our CI scripts is `yarn install`. Doing that 3 times seems wasteful. Unit tests + lint + checks takes about 1.5 mins. Uploading to Chromatic takes about 14s. Running Cypress takes a few minutes. Building Storybook takes a few minutes. We have to build storybook for both Cypress and Chromatic. It seems okay to have Cypress wait 14s for Chromatic rather than running them in parallel and deal with the verbose output of `run-p`.

Reply:
Hmm. I think I read the wrong line. Chromatic takes about 1.5 min to upload. Should we `run-p` that with Cypress? Builds are still around 10m. Running in parallel might save about 1m off that. We don't parse or annotate based off the Cypress error since there was no useful stack trace.

Cypress 4.4.0 just added support for proper stack traces: https://github.com/cypress-io/cypress/pull/6965

We could add an annotation matcher so that failed tests will show us which line failed.

Reply:
Makes sense 👍  We can probably leave parallelizing Chromatic and Cypress to a separate PR if we want to do it.